packages/design-system/src/components/InlineEditing/Primitives/InlineEditingPrimitive.tsx

```diff
@@ -84,18 +121,31 @@ const InlineEditingPrimitive = forwardRef(
 			}
 		}, [hasError]);
 
+		useEffect(() => {
+			if (onChangeValue) {
+				// Controlled mode - save value prop as snapshot
+				setSnapshotValue(value);
+			}
+		}, [onChangeValue, value]);
+
 		const handleSubmit = (event: OnEditEvent) => {
 			event.stopPropagation();
-			if (onEdit) {
-				const sentValue = value || '';
-				onEdit(event, sentValue);
+
+			if (onChangeValue) {
+				// Controlled mode - call external controller to handle validating new filled value
+				onChangeValue(inProgressValue || '');
+			} else if (onEdit) {
+				// Uncontrolled mode - save input value as snapshot and notify
+				setSnapshotValue(inProgressValue);
+				onEdit(event, inProgressValue || '');
```

Comment:
We don't know on the parent side if it's a submit or just an onChange like altering the field